flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -566,6 +592,7 @@ private void stopTaskExecutorServices() throws Exception {
 			}
 
 			if (taskAdded) {
+				submittedTasks.add(task);
```

Comment:
We should also schedule a callback to `task.getTaskCompletionFuture().thenRun(() -> submittedTasks.remove(tdd.getExecutionAttemptId()))` if `submittedTasks` is e.g. a `ConcurrentHashMap<ExecutionAttemptID, Task>`. Otherwise tasks keep piling in memory for this `TaskExecutor` for the cluster lifetime.